src/Compilers/Core/Portable/PEWriter/BlobUtilities.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.Cci
+{
+    internal unsafe static class BlobUtilities
+    {
+        public static void WriteBytes(this byte[] buffer, int start, byte value, int count)
+        {
+            fixed (byte* bufferPtr = buffer)
```

Comment:
That's a bummer. I wanted to avoid an extra range check but ```fixed``` statement it self introduces bunch of checks :(